File: Strivers A2Z DSA course/2. Learn Important Sorting Techniques/2.1. Sorting-I/BubbleSort.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Scanner;

public class BubbleSort {

    public static void bubbleSort(List<Integer> arr, int n) {
        for (int i = 0; i < n; i++) {
            boolean flag = true;
            for (int j = i; j < n - i - 1; j++) {
                if (arr.get(j + 1) < arr.get(j)) {
                    flag = false;
                    Collections.swap(arr, j, j + 1);
                }
            }
            if (flag) {
                break;
            }
        }
    }

    public static void main(String[] args) {

        int n;
        System.out.print("Enter the size of array: ");
        Scanner sc = new Scanner(System.in);
        n = sc.nextInt();
        System.out.print("Enter elements: ");
        List<Integer> arr = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            arr.add(sc.nextInt());
        }
        bubbleSort(arr, n);
        System.out.println("Sorted array: " + arr);
    }
}

File: Strivers A2Z DSA course/2. Learn Important Sorting Techniques/2.1. Sorting-I/bubble-sort.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
  
    int n;
    cout << "Enter array size: ";
    cin >> n;
    cout << "Enter array: ";
    vector<int>arr(n);
    for(int i=0; i<n; i++){
        cin >> arr[i];
    }

    for(int i=0; i<n; i++){
        bool flag = true;
        for(int j=0; j<n-i-1; j++){
            if(arr[j+1] < arr[j]){
                flag = false;
                swap(arr[j+1], arr[j]);
            }
        }
        if(flag){
            break;
        }
    }
    cout << "Sorted array: ";
    for(auto x: arr){
        cout << x << ", ";
    }
    cout << "\n";
    return 0;
}